floodlight/transforms/filter.py

```diff
@@ -0,0 +1,253 @@
+    signal_filtered = scipy.signal.filtfilt(
+        coeffs[0], coeffs[1], signal, method="pad", axis=0
+    )
+
+    return signal_filtered
+
+
+def butterworth_lowpass(
+    xy: XY,
+    order: int = 3,
+    cutoff: Numeric = 1,
+    remove_short_seqs: bool = True,
+) -> XY:
+    """Filters the position data inside a XY-object with a digital Butterworth lowpass-
+    filter [1]_. This is a wrapper for the `scipy.filter.butter
+    <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.butter.html>`_
+    and the `scipy.signal.filtfilt <https://docs.scipy.org/doc/scipy/reference/
+    generated/scipy.signal.filtfilt.html>`_ functions.
+
+    Parameters
+    ----------
+    xy: XY
+        Floodlight.XY-object.
+    order: int
+        The order of the filter. This is the argument ``N`` from the `scipy.signal.
+        butter <https://docs.scipy.org/doc/scipy/reference/generated/ scipy.signal.
+        butter.html>`_ function.
+    cutoff: Numeric
+        The critical cutoff frequency. This is the argument ``Wn`` from the
+        `scipy.signal.butter <https://docs.scipy.org/doc/scipy/reference/generated/
+        scipy.signal.butter.html>`_
+        function.
+    remove_short_seqs: bool
+        If True, sequences that are to short for the filter with the specified settings
+        are replaced with np.NaNs. If False, they are kept unfiltered. Default is True.
+
+    Returns
+    -------
+    xy_filtered: XY
+        XY object with position data filtered by designed Butterworth low pass filter.
+
+    Notes
+    -----
+    The values of the input data are assumed to be numerical. Missing data is assumed
+    to be either np.NaN or None. The Butterworth-filter requires a minimum signal length
+    depending on the settings. A signal is a sequence of data in the XY-object that is
+    not interrupted by missing values. The minimum signal length is defined as
+    :math:`3 \\cdot (order + 1)`. The treatment of signals shorter than the minimum
+    signal length are specified with the ``remove_short_sequence``-argument, where True
+    will replace these sequences with np.NaNs ond False will keep the sequences in the
+    data unfiltered.
+
+    References
+    ----------
+        .. [1] `Butterworth, S. (1930). On the theory of filter amplifiers. Wireless
+            Engineer, 3, 536-541. <https://www.changpuak.ch/electronics/downloads/
+            On_the_Theory_of_Filter_Amplifiers.pdf>`_
+    """
+
+    min_signal_len = 3 * (order + 1)
+    framerate = xy.framerate
+
+    xy_filt = np.empty(xy.xy.shape)
+    for i, column in enumerate(xy.xy.T):
+        sequences = _get_sequences_without_nans(column)
+
+        col_filt = np.full(column.shape, np.nan)
+
+        for seq in sequences[sequences[:, 2] == 1]:
+            if np.diff(seq[0:2]) > min_signal_len:
+                col_filt[seq[0] : seq[1]] = _filter_sequence_butterworth_lowpass(
+                    column[seq[0] : seq[1]], order, cutoff, framerate
+                )
```

Comment:
I think this is a solid way of performing the filter application. However, I think it can be re-formulated more explicitly, e.g. ``for start, end in sequences`` and then use ``start`` and ``end`` instead of ``seq[0]`` and ``seq[1]``. 

Also, in ``_get_sequences_without_nans(column)``, do you really need to incorporate NaN-sequences and then filter them ou there? Especially as it contradicts the function name? You could also apply the ``min_signal_len`` directly in that function, or here before the for loop